inference-engine/tests/functional/inference_engine/extension/extension.cpp

```diff
@@ -0,0 +1,422 @@
+        </layer>
+        <layer name="operation" id="1" type="Template" version="custom_opset">
+            <data  add="11"/>
+            <input>
+                <port id="1" precision="FP32">
+                    <dim>2</dim>
+                    <dim>2</dim>
+                    <dim>2</dim>
+                    <dim>1</dim>
+                </port>
+            </input>
+            <output>
+                <port id="2" precision="FP32">
+                    <dim>2</dim>
+                    <dim>2</dim>
+                    <dim>2</dim>
+                    <dim>1</dim>
+                </port>
+            </output>
+        </layer>
+        <layer name="output" type="Result" id="2" version="opset1">
+            <input>
+                <port id="0" precision="FP32">
+                    <dim>2</dim>
+                    <dim>2</dim>
+                    <dim>2</dim>
+                    <dim>1</dim>
+                </port>
+            </input>
+        </layer>
+    </layers>
+    <edges>
+        <edge from-layer="0" from-port="0" to-layer="1" to-port="1"/>
+        <edge from-layer="1" from-port="2" to-layer="2" to-port="0"/>
+    </edges>
+</net>
+)V0G0N";
+
+    std::vector<float> input_values{1, 2, 3, 4, 5, 6, 7, 8};
+    std::vector<float> expected{12, 13, 14, 15, 16, 17, 18, 19};
+    InferenceEngine::Core ie;
+    ie.AddExtension(InferenceEngine::make_so_pointer<InferenceEngine::IExtension>(get_extension_path()));
+    infer_model(ie, model, input_values, expected);
+}
+
+
+TEST(Extension, OnnxModelWithExtensionFromDSO) {
+    std::string model = R"V0G0N(
+ir_version: 3
+producer_name: "nGraph ONNX Importer"
+graph {
+  node {
+    input: "A"
+    output: "Y"
+    name: "operation"
+    op_type: "Template"
+    domain: "custom_domain"
+    attribute {
+        name: "add"
+        type: INT
+        i: 11
+    }
+  }
+  name: "test_graph"
+  input {
+    name: "A"
+    type {
+      tensor_type {
+        elem_type: 1
+        shape {
+          dim {
+            dim_value: 2
+          }
+          dim {
+            dim_value: 2
+          }
+          dim {
+            dim_value: 2
+          }
+          dim {
+            dim_value: 1
+          }
+        }
+      }
+    }
+  }
+  output {
+    name: "Y"
+    type {
+      tensor_type {
+        elem_type: 1
+        shape {
+          dim {
+            dim_value: 2
+          }
+          dim {
+            dim_value: 2
+          }
+          dim {
+            dim_value: 2
+          }
+          dim {
+            dim_value: 1
+          }
+        }
+      }
+    }
+  }
+}
+opset_import {
+  version: 1
+  domain: "com.example"
+}
+)V0G0N";
+
+    std::vector<float> input_values{1, 2, 3, 4, 5, 6, 7, 8};
+    std::vector<float> expected{12, 13, 14, 15, 16, 17, 18, 19};
+    InferenceEngine::Core ie;
+    ie.AddExtension(InferenceEngine::make_so_pointer<InferenceEngine::IExtension>(get_extension_path()));
+    infer_model(ie, model, input_values, expected);
+}
```

Comment:
@ilyachur I dislike the fact that this test infers model via CPU plugin. I think we need to compare the read network with reference network as it's done in tests for IR v10 reader, transformations tests.

These tests for IE only, we should not use any plugins here.

Reply:
These tests check the full pipeline.
I means that we get a correct model, register the custom operation and custom execution kernel for CPU and infer it.
The reason why we need to infer these tests only for CPU is that other plugins don't register custom kernels from Extension API.